pip/locations.py

```diff
@@ -184,9 +185,13 @@ def distutils_scheme(dist_name, user=False, home=None, root=None,
     # NOTE: setting user or home has the side-effect of creating the home dir
     # or user base for installations during finalize_options()
     # ideally, we'd prefer a scheme class that has no side-effects.
+    if user and prefix:
+        raise InstallationError(
+            "Can not combine '--user' and '--prefix' as they imply different "
+            "installation locations"
+        )
```

Comment:
Hmm I was thinking of `CommandError` raised in `pip/commands/install.py` since `--user` and `--prefix` are two incompatible command line options.
An assert here in `def distutils_scheme` could recheck that everything is fine.